### DISCUSSION TITLE
Working on Microcontroller?

### DISCUSSION THREAD

**Discussion by D-c8** [serizba/cppflow#177] [Q&A]
Is there a way this works on a microcontroller without GPU?

**Comment by serizba** [comment]
As long as you can run TensorFlow you should be able to run cppflow on top of it. GPU is not a requirement, just download the CPU version from the TF C API [site](https://www.tensorflow.org/install/lang_c).

**Reply by D-c8** [reply]
Thank you for the extremely fast reply!

**Reply by D-c8** [reply]
Also, do you have data of how much memory it uses? I just want to make sure the library does not overextend the ressources of a microcontroller.

**Reply by serizba** [reply]
Memory requirements will mainly depend on what model you want to run with TensorFlow. Cppflow is a lightweight wrapper, so it doesn't add much overhead. Most memory consumption will be from TensorFlow.
